### PR TITLE
cxx-qt-gen: use namespace = "A::B" from cxx_qt::bridge

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Requires #193

Note that currently each QObject must be in it's own namespace otherwise free functions will collide. This will be improved in the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ impl From<Data> for DataSerde {
     }
 }
 
-#[cxx_qt::bridge]
+#[cxx_qt::bridge(namespace = "cxx_qt::my_object")]
```
Comment (ahayzen-kdab) [inline README.md:55]: Need to document the namespace in the book and state how currently each QObject needs it's own namespace otherwise there will be collisions until we get a bit further....

Comment (ahayzen-kdab) [reply README.md:55]: Added some minimal documentation around this for now.

File: cxx-qt-gen/src/syntax/attribute.rs [hunk 2]
```diff
@@ -136,4 +186,48 @@ mod tests {
         assert_eq!(attribute_tokens_to_list(&module.attrs[4]).unwrap()[1], "B");
         assert_eq!(attribute_tokens_to_list(&module.attrs[4]).unwrap()[2], "C");
     }
+
+    #[test]
+    fn test_attribute_tokens_to_map() {
+        let module: ItemMod = tokens_to_syn(quote! {
```
Comment (ahayzen-kdab) [inline cxx-qt-gen/src/syntax/attribute.rs:192]: need to add `#[attribute()]` as a test case to check that an empty map is considered (and check if the list tests has it too).

Comment (ahayzen-kdab) [reply cxx-qt-gen/src/syntax/attribute.rs:195]: Fixed and added some tests in mod.rs for namespace too.

File: book/src/qobject/macro.md [hunk 3]
```diff
@@ -7,7 +7,11 @@ SPDX-License-Identifier: MIT OR Apache-2.0
 
 # Macro
 
-We define a module (which becomes our Qt object name) and then add `cxx_qt::bridge` as a macro.
+We define a module (which becomes our Qt object name) and then add `cxx_qt::bridge(namespace = "cxx_qt::my_object)` as a macro.
+
+The namespace specified is used in C++ to allow segmenting generated code from your application.
+
+Note that currently each QObject needs to be in it's own namespace otherwise there will be collisions from generated free functions.
```
Comment (Be-ing) [inline book/src/qobject/macro.md:14]:
```suggestion
Note that currently each QObject needs to be in its own namespace otherwise there will be collisions from generated free functions.
```

Comment (ahayzen-kdab) [reply book/src/qobject/macro.md]: Fixed

File: cxx-qt-gen/src/gen_rs.rs [hunk 4]
```diff
@@ -478,11 +359,8 @@ pub fn generate_qobject_cxx(
     // Build the import path for the C++ header
     let import_path = format!("cxx-qt-gen/include/{}.cxxqt.h", ident_snake);
 
-    // TODO: ideally we only want to add the "type QString = cxx_qt_lib::QString;"
-    // if we actually generate some code that uses QString.
-
     // Build the namespace string, rust::module
```
Comment (Be-ing) [inline cxx-qt-gen/src/gen_rs.rs:362]: comment no longer needed? I think the code is self-explanatory now

File: cxx-qt-gen/src/syntax/attribute.rs [hunk 5]
```diff
@@ -135,5 +186,55 @@ mod tests {
...
+            #[invokable]
+            #[cxx_qt::bridge]
+            #[cxx_qt::signals(MyObject)]
+            #[cxx_qt::bridge(namespace = "my::namespace")]
+            #[cxx_qt::list(A, B, C)]
+            #[cxx_qt::bridge(a = "b", namespace = "my::namespace")]
+            #[cxx_qt::bridge(a = "b", namespace = "my::namespace", namespace = "my::namespace")]
```
Comment (Be-ing) [inline cxx-qt-gen/src/syntax/attribute.rs:202]: I know this is just testing the syntax parsing, but in the actual parser I think it should return an error if multiple namespaces are specified.

File: cxx-qt/src/lib.rs [hunk 6]
```diff
@@ -37,9 +37,6 @@ pub fn bridge(args: TokenStream, input: TokenStream) -> TokenStream {
     // this triggers a compile failure if the tokens fail to parse.
     let mut module = parse_macro_input!(input as ItemMod);
 
-    // Macros do not typically need to do anything with their own attribute name,
-    // so rustc does not include that in the `args` or `input` TokenStreams.
-    //
```
Comment (Be-ing) [inline cxx-qt/src/lib.rs:42]: Was deleting this an accident when resolving conflicts?

File: cxx-qt-gen/src/syntax/attribute.rs (hunk 5 above)
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/syntax/attribute.rs:202]: This test is checking that that case errors, `assert!(attribute_tokens_to_map::<Ident, LitStr>(&module.attrs[6]).is_err());`

File: cxx-qt-gen/src/gen_rs.rs (hunk 4 above)
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/gen_rs.rs]: Removed.

File: cxx-qt/src/lib.rs (hunk 6 above)
Comment (ahayzen-kdab) [reply cxx-qt/src/lib.rs]: Oops, yup, reverted.
